src/pc/common/BitArray.js

```diff
@@ -134,6 +134,7 @@ class BitArray {
   readBuffer (smartBuffer, size = this.data.length) {
     if (size !== this.data.length) {
       this.data = new Uint32Array(size)
+      this.readOffset = size * 4
```

Comment:
This is crazy surprising, I don't think this belongs here.

If I call `readBuffer` a read should always happen. I might be missing context

Reply:
Check my comment here https://github.com/PrismarineJS/prismarine-chunk/pull/232#issuecomment-1693890460

Reply:
And for additional context see https://github.com/extremeheat/extracted_minecraft_data/blob/20e005fafc4928179f144d3ff68413f0ca70808b/client/net/minecraft/world/level/chunk/PalettedContainer.java#L163

Reply:
I don't see how return'ing early from `readBuffer` after skipping the data that was requested to be read has anything to do with the linked code.

Reply:
To summarize what is happening in the Vanilla implementation:

The Vanilla client attempts to read a long array into the existing raw array in the ChunkStorage. The readLongArray function then checks to see if the size of the existing array is equal to the size of the network provided long array (which is what I am doing with `size !== this.data.length` here). If it is not equal, a new long array will be created locally and returned from the function, but the return value of readLongArray is not used in the PalettedContainer read method, so the result is simply discarded.

My implementation technically doesn't mirror this behaviour as it doesn't create a new long array, but the result is the same, just less wasteful.

Reply:
Thank you for explaining